internal/services/domainservices/resourceids.go

```diff
@@ -2,3 +2,5 @@ package domainservices
 
 //go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=DomainService -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.AAD/domainServices/DomainService1/initialReplicaSetId/replicaSetID
 //go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=DomainServiceReplicaSet -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.AAD/domainServices/DomainService1/replicaSets/replicaSetID
+//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=DomainServiceTrust -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.AAD/domainServices/DomainService1/trusts/trust1
+//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=DomainServiceResource -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.AAD/domainServices/DomainService1
```

Comment:
This resource is already defined with existing parsers/validators as `DomainService`. To avoid confusion, I think it would be better to parse that, removing the custom suffix as needed?

Reply:
The suffix (`trusts/trust1`) is needed as there can be multiple trusts configured for a AADDS, the `trust1` corresponds to the friendly name of the trust.

Reply:
Sure, but I meant the DomainServiceResource ID represents the same thing as the DomainService ID

Reply:
Currently, `DomainServiceResource` is only used for logging.. I can remove it then. Whilst, I'm wondering why did we choose the long form `/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.AAD/domainServices/DomainService1/initialReplicaSetId/replicaSetID` to represent the AADDS, rather than the `DomainServiceResource` form at the first place? As the former is hard for users to identify (imagin when users want to import it), and also depends on the ordering of the replica sets.

Reply:
We use a contrived ID that includes the initial replica set in order to overcome a conflict of interest between Terraform and the API - we have to keep track of the replica set that is implicitly created with the resource, so that we can also manage additional replica sets.

Reply:
It has been removed now.